pkg/porter/dependencies.go

```diff
@@ -250,20 +252,27 @@ func (e *dependencyExecutioner) executeDependency(dep *queuedDependency, parentA
 		CredentialIdentifiers: parentArgs.CredentialIdentifiers,
 	}
 
+	// Determine if we're working with UninstallOptions, to inform deletion and
+	// error handling, etc.
+	var uninstallOpts UninstallOptions
+	if opts, ok := e.parentOpts.(UninstallOptions); ok {
+		uninstallOpts = opts
+	}
+
 	var executeErrs error
 	fmt.Fprintf(e.Out, "Executing dependency %s...\n", dep.Alias)
 	err := e.CNAB.Execute(depArgs)
 	if err != nil {
 		executeErrs = multierror.Append(executeErrs, errors.Wrapf(err, "error executing dependency %s", dep.Alias))
 
 		// Handle errors when/if the action is uninstall
-		executeErrs = e.parentOpts.handleUninstallErrs(e.Out, executeErrs)
+		executeErrs = uninstallOpts.handleUninstallErrs(e.Out, executeErrs)
 		if executeErrs != nil {
 			return executeErrs
 		}
 	}
 
-	if e.parentOpts.shouldDelete() {
+	if uninstallOpts.shouldDelete() {
```

Comment:
If uninstallOpts is an empty struct (because parentOpts was some other action), what does this call do?

Reply:
It *think* all methods on an empty struct (when action not uninstall) return the uninitialized/default values for the struct fields... so mostly boolean `false` values, which is what we'd want here.  The tests seem to agree.

However, I can see this approach/code being fragile... even if the assumptions are true.  What do we think?  Should we refactor the areas where we rely on uninstallOpts to run only if we are sure the struct is actually that type (and not empty)?

Reply:
I think it would get pretty gnarly to conditionally call those, so the empty struct is fine. I was just double checking. Maybe let's just add a comment about if it's not uninstall, it's a no-op handler?